File: scripts/shared/finding-format.mjs
#!/usr/bin/env node

import { normalizeFindingId } from "./finding-id.mjs";

function normalizeInline(value) {
  return String(value ?? "")
    .replace(/\r?\n+/g, " ")
    .replace(/\s{2,}/g, " ")
    .trim();
}

export function formatFindingHeadline({ reviewerLabel, finding }) {
  const findingId = normalizeInline(normalizeFindingId(finding?.id || ""));
  const title = normalizeInline(finding?.title || "Untitled finding");
  if (findingId) {
    return `**[${findingId}]** ${title}`;
  }
  const normalizedReviewerLabel = normalizeInline(reviewerLabel || "Unknown Reviewer");
  return `**${normalizedReviewerLabel}:** ${title}`;
}

export function formatFindingRecommendation(finding) {
  return normalizeInline(finding?.recommendation || "No recommendation provided.");
}

export function formatFindingBody({ reviewerLabel, finding }) {
  return [
    formatFindingHeadline({ reviewerLabel, finding }),
    "",
    formatFindingRecommendation(finding),
  ].join("\n");
}

File: scripts/shared/finding_format.py
from __future__ import annotations

from .finding_id import normalize_finding_id
from .types import PresentationFinding


def _normalize_inline(value: str | int | float | bool | None) -> str:
    return " ".join(str("" if value is None else value).replace("\n", " ").split())


def format_finding_headline(*, reviewer_label: str, finding: PresentationFinding) -> str:
    finding_id = _normalize_inline(normalize_finding_id(finding.get("id")))
    title = _normalize_inline(finding.get("title", "Untitled finding"))
    if finding_id:
        return f"**[{finding_id}]** {title}"

    normalized_reviewer = _normalize_inline(reviewer_label or "Unknown Reviewer")
    return f"**{normalized_reviewer}:** {title}"


def format_finding_recommendation(finding: PresentationFinding) -> str:
    return _normalize_inline(finding.get("recommendation", "No recommendation provided."))


def format_finding_body(*, reviewer_label: str, finding: PresentationFinding) -> str:
    return "\n".join(
        [
            format_finding_headline(reviewer_label=reviewer_label, finding=finding),
            "",
            format_finding_recommendation(finding),
        ]
    )
